### PR TITLE
Pm 2020 04 bootstrap comp rebuilds

### PR DESCRIPTION
**Pull request by phillimorland**
ended up rebuilding and refactoring all the code then they worked :D i'm still not totally sure what exactly was making it not work though?!

### CODE REVIEW

File: euth/projects/templates/euth_projects/project_detail.html [hunk 1]
```diff
@@ -31,23 +31,27 @@
...
+                      data-placement="top"
+                      id="button-phase-{{ forloop.counter }}"
+                      data-content="
+                      <p class='small'>{{ item.start_date|date:"SHORT_DATETIME_FORMAT" }} - {{ item.end_date|date:"SHORT_DATETIME_FORMAT" }}</p>
+                      <h4>{{ item.name }}</h4>
+                    </button>
+
+                    {% if item.id in ids or item.id == view.project.active_phase.id %}
+                    -                    <a href="{% url 'project-detail' slug=view.project.slug %}?phase={{ item.weight }}">
+                    -                    {% endif %}
```
Comment (fuzzylogic2000) [inline euth/projects/templates/euth_projects/project_detail.html]: Are the "-" here on purpose?

File: euth/projects/templates/euth_projects/project_detail.html [hunk 2]
```diff
@@ -32,26 +32,30 @@
...
-                      data-trigger="hover | click"
+                      data-trigger="hover | focus"
                       data-html="true"
                       data-placement="top"
                       id="button-phase-{{ forloop.counter }}"
                       data-content="
                       <p class='small'>{{ item.start_date|date:"SHORT_DATETIME_FORMAT" }} - {{ item.end_date|date:"SHORT_DATETIME_FORMAT" }}</p>
-                      <h4>{{ item.name }}</h4>
+                      <h4>{{ item.name }}</h4>"
+                      >
```
Comment (fuzzylogic2000) [inline euth/projects/templates/euth_projects/project_detail.html:53]: Should this be here?

File: euth/projects/templates/euth_projects/project_detail.html [hunk 3]
```diff
@@ -62,15 +66,19 @@ <h4>{{ item.name }}</h4>
...
+                         data-toggle="popover"
+                         data-html="true"
+                         data-trigger="hover | click | focus"
+                         data-placement="top"
                          id="button-phase-{{ forloop.counter }}"
                          data-content="
                                 <p class='small'>{{ item.date|date:"SHORT_DATETIME_FORMAT" }}</p>
                                 <h4>{{ item.name }}</h4>
-                            "></div>
+                            ">
```
Comment (fuzzylogic2000) [inline euth/projects/templates/euth_projects/project_detail.html:80]: And this? Am I overlooking sth.?

File: euth/projects/templates/euth_projects/project_detail.html (hunk 2 above)
Comment (phillimorland) [reply euth/projects/templates/euth_projects/project_detail.html:53]: yes, its a thing with popovers, they have the html template within the data-content attribute

**Pull request merged**